includes/class-search-categories-metabox.php

```diff
@@ -0,0 +1,326 @@
+<?php
+
+class Search_Categories_Metabox {
+
+    /**
+     * @const PLUGIN_SLUG
+     */
+    CONST PLUGIN_SLUG = 'search-categories-metabox';
+
+    /**
+     * @const TEXT_DOMAIN
+     */
+    CONST TEXT_DOMAIN = 'search-categories-metabox';
+
+    /**
+     * @var string $plugin_file
+     */
+    protected $plugin_file;
+
+    /**
+     * @var string $plugin_version
+     */
+    public $plugin_version = '0.1.0';
+
+    /**
+     * Search_Categories_Metabox constructor.
+     * @param $file
+     */
+    function __construct( $file ) {
+        $this->plugin_file = $file;
+    }
+
+    /**
+     * Gets the Post taxonomy Category slug by Post ID.
+     *
+     * @since 0.1.0
+     *
+     * @param $post_id
+     * @return array
+     */
+    function _get_post_taxonomy( $post_id ) {
```

Comment:
It's a matter of personal preference, but I would prefer to explicitly state access modifiers like `public` or `private`, instead of using conventions like `_`.